elfi/methods/bo/acquisition.py

```diff
@@ -622,11 +623,7 @@ def __init__(self, quantile_eps=.01, integration='grid', d_grid=.2,
                                          n_samples=n_samples,
                                          sigma_proposals_metropolis=sigma_proposals_metropolis)
         elif self._integration == 'grid':
-            slices_param = ()
-            for bound in self.model.bounds:
-                slice_param = slice(bound[0], bound[1], d_grid)
-                slices_param = slices_param + (slice_param,)
-            grid_param = list(slices_param)
+            grid_param = [slice(b[0], b[1], d_grid) for b in self.model.bounds]
             self.points_int = np.mgrid[grid_param].reshape(len(self.model.bounds), -1).T
```

Comment:
Can't follow the code here (because my lack of python skills), I guess we select some amount of points for each coordinate based on the grid element length d_grid and construct then a grid with (grid elements)^dim amount of grid points? This is ok for dim <= 3 (I used similar grid strategy but used it for 1d/2d cases in my paper). I wonder if we should disallow using grid in high dim, the user might set too many points? Or could you code some quasi-Monte carlo sequence (see https://en.wikipedia.org/wiki/Low-discrepancy_sequence, the figures after all that math...), There must be readily available code for obtaining such grid points in numpy or elsewhere?

Reply:
In the doc string, it is mentioned that the grid sampling is computationally expensive in high dimensions. Additional input checks would just repeat what is already stated.

Reply:
ok, I agree, I was just wondering what might happen if the user nevertheless tries to use this approach in - say - 10d case? -> Crash or other issue with too high memory usage? This would be users fault then of course, so maybe we can let this be like this.

Reply:
Under a reasonable number of grid points, the integration would become computationally infeasible. If there was no crash associated with memory, the user would forever see the ELFI's interface which would not stop executing.